File: Frontend appbrc/brandcomputers_v1.1/src/app/model/components/Case.ts
import {GenerateProductCode} from './GenerateProductCode';

export class Case {
    id: number;
    generateProductCode: GenerateProductCode;
    serialNumber: string;
    manufacturer: string;
    caseType: string;
    compatibleMotherboard: string;
    powerSourceIncluded: boolean;
    sourcePower: string;
    priceIn: number;
    productInformation: string;
    state: string;
    soldAt: Date;
    soldBy: string;
    sold: boolean;
    createdDate: Date;
    createdBy: string;
    lastUpdated: Date;
    updatedBy: string;
    quantity: number;
    unitOfMeasurement: string;

    constructor(obj) {
        this.id = obj.id;
        this.generateProductCode = obj.generateProductCode;
        this.serialNumber = obj.serialNumber;
        this.manufacturer = obj.manufacturer;
        this.caseType = obj.caseType;
        this.compatibleMotherboard = obj.compatibleMotherboard;
        this.powerSourceIncluded = obj.powerSourceIncluded;
        this.sourcePower = obj.sourcePower;
        this.priceIn = obj.priceIn;
        this.productInformation = obj.productInformation;
        this.state = obj.state;
        this.soldAt = obj.soldAt;
        this.soldBy = obj.soldBy;
        this.sold = obj.sold;
        this.createdDate = obj.createdDate;
        this.createdBy = obj.createdBy;
        this.lastUpdated = obj.lastUpdated;
        this.updatedBy = obj.updatedBy;
        this.quantity = obj.quantity;
        this.unitOfMeasurement = obj.unitOfMeasurement;
    }
}

File: Backend appbrc/brandcomputer/src/main/java/com/app/brc/brandcomputer/components/casing/model/Case.java
package com.app.brc.brandcomputer.components.casing.model;

import com.app.brc.brandcomputer.components.product_code.model.ProductCode;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Builder
@ToString
@Entity
@Table(name = "cases")
public class Case {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "productCode_id")
    private ProductCode generateProductCode;

    private String serialNumber;
    private String manufacturer;
    private String caseType;
    private String compatibleMotherboard;
    private boolean powerSourceIncluded;
    private String sourcePower;
    private Double priceIn;
    private String productInformation;
    private String state;
    private String category;
    private int quantity;
    private String unitOfMeasurement;
    private LocalDateTime soldAt;
    private String soldBy;
    private boolean sold;
    private boolean received;

    @CreationTimestamp
    private LocalDateTime createdDate;
    private String createdBy;

    @UpdateTimestamp
    private LocalDateTime lastUpdated;
    private String updatedBy;

}
